Source/santad/EventProviders/EndpointSecurity/Enricher.mm

```diff
@@ -81,26 +81,30 @@
   }
 }
 
-EnrichedProcess Enricher::Enrich(const es_process_t &es_proc) {
-  return EnrichedProcess(UsernameForUID(audit_token_to_euid(es_proc.audit_token)),
-                         UsernameForGID(audit_token_to_egid(es_proc.audit_token)),
-                         UsernameForUID(audit_token_to_ruid(es_proc.audit_token)),
-                         UsernameForGID(audit_token_to_rgid(es_proc.audit_token)),
-                         Enrich(*es_proc.executable));
+EnrichedProcess Enricher::Enrich(const es_process_t &es_proc, EnrichOptions options) {
+  return EnrichedProcess(UsernameForUID(audit_token_to_euid(es_proc.audit_token), options),
+                         UsernameForGID(audit_token_to_egid(es_proc.audit_token), options),
+                         UsernameForUID(audit_token_to_ruid(es_proc.audit_token), options),
+                         UsernameForGID(audit_token_to_rgid(es_proc.audit_token), options),
+                         Enrich(*es_proc.executable, options));
 }
 
-EnrichedFile Enricher::Enrich(const es_file_t &es_file) {
+EnrichedFile Enricher::Enrich(const es_file_t &es_file, EnrichOptions options) {
   // TODO(mlw): Consider having the enricher perform file hashing. This will
   // make more sense if we start including hashes in more event types.
-  return EnrichedFile(UsernameForUID(es_file.stat.st_uid), UsernameForGID(es_file.stat.st_gid),
-                      std::nullopt);
+  return EnrichedFile(UsernameForUID(es_file.stat.st_uid, options),
+                      UsernameForGID(es_file.stat.st_gid, options), std::nullopt);
 }
 
-std::optional<std::shared_ptr<std::string>> Enricher::UsernameForUID(uid_t uid) {
+std::optional<std::shared_ptr<std::string>> Enricher::UsernameForUID(uid_t uid,
+                                                                     EnrichOptions options) {
   std::optional<std::shared_ptr<std::string>> username = username_cache_.get(uid);
 
   if (username.has_value()) {
     return username;
+  } else if (options == EnrichOptions::kLocalOnly) {
```

Comment:
maybe this would be better named `kCacheOnly`? to me local only sounds like it would read out of /etc/passwd or similar (but not cause processes to spawn or network accesses), but this only returns if something had previously set it in our local cache.

Reply:
Maybe `kNoExternalLookups`? The key is meant to describe that any enrichment performed must not reach out to external processes.

Reply:
Ah, ok. In that case I think it's fine to leave as-is - just wasn't sure if it was always going to be "cache only" or may in the future read locally.